### ISSUE TITLE
a more complete USAGE README

### ISSUE THREAD

**Issue by hanm2019** [TeCSAR-UNCC/gem5-SALAM#6]
dear, I tried to use the gem5-SALAM, after reading the README and documents in doc, I still confused about how to use it, such as the `ini` file and the `pragma`, i don't know which item needed adding in ini file and the meaning of it, and for the `pragma`, it is better to have a list which can be used in program

**Comment by Jsslycord** [comment]
Hello,
We are in the process of reworking many aspects of gem5-SALAM to be more user friendly, including automating setup file generation, combining all knobs into a single easier to use configuration, and creating an LLVM api wrapper around our execution engine. Our primary goal was to create a new tool for design space exploration and validating its accuracy, which we have achieved. Now that we have confirmed our execution model (see our MICRO 2020 publication), we are totally redesigning our interfaces for ease of use for the end user. The next working version should be available soon, so keep checking in, and thanks for your comments.
